src/posix/platform/infra_if.cpp

```diff
@@ -231,15 +236,49 @@ void platformInfraIfInit(otInstance *aInstance, const char *aIfName)
     rval = setsockopt(sock, IPPROTO_IPV6, IPV6_MULTICAST_HOPS, &kHopLimit, sizeof(kHopLimit));
     VerifyOrDie(rval == 0, OT_EXIT_ERROR_ERRNO);
 
+    return sock;
+}
+
+uint32_t platformInfraIfInit(otInstance *aInstance, const char *aIfName)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    ssize_t  rval;
+    uint32_t ifIndex = 0;
+
+    if (strlen(aIfName) >= sizeof(sInfraIfName))
+    {
+        otLogCritPlat("infra interface name '%s' is too long", aIfName);
+        DieNow(OT_EXIT_INVALID_ARGUMENTS);
+    }
+    strcpy(sInfraIfName, aIfName);
+
+    // Initializes the infra interface.
+    ifIndex = if_nametoindex(aIfName);
+    if (ifIndex == 0)
+    {
+        otLogCritPlat("failed to get the index for infra interface %s", aIfName);
+        DieNow(OT_EXIT_INVALID_ARGUMENTS);
+    }
+    sInfraIfIndex = ifIndex;
+
+    // TODO: We want to verify if there are IPv6 link-local address at starting.
```

Comment:
Is this difficult to address in this PR?

Reply:
Thanks for catching this, this is a stale TODO which is addressed by after refactoring.